addons/web/static/src/core/assets.js

```diff
@@ -108,23 +108,45 @@ export const loadBundleTemplates = memoize(async function loadBundleTemplates(na
     return processTemplates(templates);
 });
 
+
+export const _loadBundle= memoize(async function(bundleName){
+    const bundleURL = new URL(`/web/bundle/${bundleName}/jscss`, window.location.origin);
+    const bundleInfo = await (await browser.fetch(bundleURL.href)).json();
+    /*[{"type": "script", "src": "/web/assets/266-d34b0b4/documents_spreadsheet.o_spreadsheet.min.js", "mimetype": "text/javascript"}]*/
+    return bundleInfo;
+});
+
 const bundlesCache = {};
+
+
 /**
  * Loads a bundle.
  *
  * @param {string} name the name of the bundle to load
  * @param {Bundle} options parts of the bundle to load (see Bundle typedef)
  * @returns {Promise<LoadedBundle>}
  */
-async function loadBundle(name, options) {
+export async function loadBundle(name, options) {
     const { templates } = options;
     if (!bundlesCache[name]) {
         bundlesCache[name] = { name };
     }
     if (templates && !bundlesCache[name].templates) {
         bundlesCache[name].templates = loadBundleTemplates(name);
     }
-    // TODO: if ("js/css") {...} to support lazy loading js/css from bundles
+
+    if (options.css || options.js) {
+        const bundleInfo = await _loadBundle(name);
+        const scriptInfo = bundleInfo.find((i)=> i.type ==="script")
+        if (options.js && scriptInfo) {
+            loadJS(scriptInfo.src);
```

Comment:
should probably store this one in the bundlesCache (the css too). That will also have the advantage that the returned promise will properly wait for everything that was requested to be loaded. With the current setup the only way to know that the js or css of the requested bundle is loaded is by observing side-effects of the load (presence of global variables or of css rules)